pages/Messages/Validation.md

```diff
@@ -0,0 +1,117 @@
+---
+name: Validation
+route: /Messages/Validation
+menu: Messages
+---
+
+# Message Validation
+
+All messages must be validated by announcers before being included in a batch file, however clients and indexers should not rely on announcer validation alone and should perform their own validation on incoming messages as well.
+Message validation in this document is defined as a collection of independent checks which may be run in parallel depending on implementation, however all checks are required to pass for a message to be considered valid.
+
+In general, all implementations must validate announcements for correctness against the [DSNP message schema](/Messages/Overview) and authenticity using [their provided signatures](/Messages/Signatures).
+Client and indexer implementations must also validate the activity pub content of applicable messages for correctness with the [Activity Pub schema](https://www.w3.org/TR/activitypub/) and authenticity using the provided content hash.
+Announcers may choose to skip content validation checks in the interest of performance given the high cost of fetching content.
+
+## Specification Status
+
+| Version | Status |
+---------- | ---------
+| 0.1     | Tentative |
+
+## Purpose
+1. Describe the process of validating messages for inclusion in a batch
+
+## Assumptions
+* All assumptions from [DSNP Messages](/Messages/Overview)
+* All assumptions from [DSNP Signatures](/Messages/Signatures)
+
+## Announcement Correctness
+
+Validation of announcement correctness will vary depending on which fields are present on a particular DSNP message type.
+Validating correctness for a message will consist of verifying that each field in the message is defined and meets the formatting rules listed in the subsections below for the given field type.
+
+#### Addresses
+
+1. Address fields must be exactly 20 bytes.
```

Comment:
I actually don't know if we have any addresses anymore as we are changing `fromAddress` to be `dsnpId` (which is just 64 bits (8 bytes))

Reply:
ok, I'll drop this section and add a "User Id" section.